docs/src/writing_good_rules.md

```diff
@@ -175,6 +176,68 @@ end
 ```
 to define the rules.
 
+## Ensure your pullback can accept the right types
+As a rule the number of types you need to accept in a pullback is theoretically unlimitted, but practically highly constrained to inline with the primal return type.
+The three kinds of inputs you will practically need to accept one or more of: _natural tangents_, _structural tangents_, and _thunks_.
+You do not in general have to handle `AbstractZero`s as the AD system will not call the pullback if the input is a zero, since the output will also be.
+Some more background information on these types can be found in [the design notes](@ref manytypes).
+In many cases these all all tangents can be treated the same: tangent types overload a bunch of linear-operators, and the majority of functions used inside a pullback are linear operators.
+If you find linear operators from Base/stdlibs that are not supported, consider openning an issue or PR on the [ChainRulesCorejl repo](https://github.com/JuliaDiff/ChainRulesCore.jl/).
+
+### Natural tangents
+Natural tangent types are the types you might feel the tangent should be.
+These are a purely human notion, they are the types the user wants to use because they make the math easy.
```

Comment:
I think this is a bit vague. Even if you don't provide an all-encompassing definition, you need to tell the reader roughly what you mean by "natural" at all, what distinction are you drawing? If they don't already know then all these disclaimers are pretty confusing.

I would do this by example. The simplest one would be `ComplexF64`, that's a struct, and by "natural" you mean a gradient which is another number, as opposed to "structural" you mean `Tangent{Complex}(...)`. And after that, move on to more complicated ones.

Reply:
Are there not several examples in the next section?

Reply:
There are, but you don't quite spell out what this word "natural" is being used to mean, before using it a bunch of times. I think that should go first: first define the terminology, then the clear-cut cases on both sides, then the tricky cases.

Reply:
ok fair.